src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.cpp

```diff
@@ -27,14 +27,23 @@
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/IR/DerivedTypes.h"
 #include <cstdint>
+#include <src/Dialect/Mlir/IndexExpr.hpp>
 
 using namespace mlir;
 namespace onnx_mlir {
 namespace tosa {
 
-Value sliceTensor(PatternRewriter &rewriter, Operation *op,
-    Value &inputConst, const llvm::ArrayRef<int64_t> &size,
-    const llvm::ArrayRef<int64_t> &start) {
+llvm::SmallVector<int64_t> createInt64VectorFromIndexExpr(
+    llvm::ArrayRef<IndexExpr> indexVector) {
+  llvm::SmallVector<int64_t, 4> literalVector;
+  for (const auto &indexExpr : indexVector) {
+    literalVector.push_back(indexExpr.getLiteral());
+  }
```

Comment:
I'm not sure I got the syntax right, but the use of `llvm::transform` is preferable to such loops.
```suggestion
llvm::transform(indexVector, literalVector.begin(), [](const auto &indexExpr) { indexExpr.getLiteral(); });
```